### PR TITLE
Add URI checking and Capabilities checking, small bug tweaks

### PR DESCRIPTION
**Pull request by tomasg2012**
The tool can now check URIs on any service at and above version 1.6.0, testing all @odata.id possible on each Resource.

Now tests headers to check if Capabilities is properly applied to the Allow header.

Small tweaks

MinVersion is now lessthan or equal

Nav properties now have RedfishType.IsNav

Fix modified results using `success` rather than `result`

ex:
image

EDIT:

Changed Error to Warn on PATCH/PUT header

Renamed variables, made URI checking less strict

Loosened URI pattern regex, Pound sign check for fragments

Change fulltype variable, proper check for incorrect non-dictionary RedfishObject values

### CODE REVIEW

File: common/catalog.py [hunk 1]
```diff
@@ -794,8 +866,59 @@ def populate(self, payload, check=False, casted=False):
...
 
+            # Validate our Uri
+            sub_obj.HasValidUri = True
+            my_uris = sub_obj.Type.getUris()
+            # If we have expected URIs and @odata.id
+            # And we AREN'T a navigation property
+            if not sub_obj.Type.catalog.flags['ignore_uri_checks'] and len(my_uris) and '@odata.id' in load:
+                # Strip our URI and warn if that's the case
+                my_odata_id = load['@odata.id']
+                if my_odata_id != '/redfish/v1/' and my_odata_id[-1] == '/':
```
Comment (mraineri) [inline common/catalog.py:880]: Long term we might not want to have trailing slash removal; need to review with others, but won't hold up this PR (more of a thought exercise for interop testing later).

File: validateResource.py [hunk 2]
```diff
@@ -133,6 +137,35 @@ def validateSingleURI(service, URI, uriName='', expectedType=None, expectedJson=
...
+    if response and response.getheader('Allow'):
+        allowed_responses = [x.strip().upper() for x in response.getheader('Allow').split(',')]
+        if not redfish_obj.Type.CanInsert and 'POST' in allowed_responses:
+            my_logger.error('Allow header should NOT contain POST for {}'.format(redfish_obj.Type))
+            counts['failAllowHeader'] += 1
+        if not redfish_obj.Type.CanDelete and 'DELETE' in allowed_responses:
+            my_logger.error('Allow header should NOT contain DELETE for {}'.format(redfish_obj.Type))
+            counts['failAllowHeader'] += 1
+        if not redfish_obj.Type.CanUpdate and any([x in allowed_responses for x in ['PATCH', 'PUT']]):
+            my_logger.error('Allow header should NOT contain PATCH or PUT for {}'.format(redfish_obj.Type))
```
Comment (mraineri) [inline validateResource.py:167]: Downgrade this to a warning; in the future, we might want to consider having dynamic checking for PATCHable OEM extensions.

**Comment by mraineri** [comment]
Will hold off until more testing done; overall this looks good though.

**Comment by tomasg2012** [comment]
Closing due to signing issue, see next PR...

**Pull request closed without merging**